regnety/dataset/imagenet.py

```diff
@@ -80,159 +80,145 @@ def decode_example(self, example: tf.Tensor) -> dict:
             "synset": synset,
         }
 
+
     def _read_tfrecs(self) -> Type[tf.data.Dataset]:
         """Function for reading and loading TFRecords into a tf.data.Dataset.
-
         Returns:
             A tf.data.Dataset
         """
-        ds = tf.data.TFRecordDataset(self.tfrecs_filepath)
-        ds = ds.map(
-            lambda example: tf.io.parse_example(example, _TFRECS_FORMAT),
-            num_parallel_calls = tf.data.AUTOTUNE
-        )
+
+        files = tf.data.Dataset.list_files(self.tfrecs_filepath)
+
+        options = tf.data.Options()
+        options.experimental_deterministic = False
+
+        files = files.with_options(options)
+
+        ds = files.interleave(tf.data.TFRecordDataset, 
+          num_parallel_calls = tf.data.AUTOTUNE,
+          deterministic=False)
+
         ds = ds.map(
-            lambda example: self.decode_example(example), 
+            self.decode_example, 
             num_parallel_calls = tf.data.AUTOTUNE 
         )
+
+        ds = ds.cache('tf_cache')
+       
+        ds = ds.repeat()
+        ds = ds.batch(self.batch_size)
+        ds = ds.prefetch(tf.data.AUTOTUNE)
         return ds
 
-    def _scale_and_center_crop(self, 
-        image: tf.Tensor,
-        scale_size: tf.Tensor, 
-        final_size: tf.Tensor) -> tf.Tensor:
-        """Resizes image to given scale size and returns a center crop. Aspect
-        ratio is maintained. Note that final_size must be less than or equal to
-        scale_size.
+  
+    def _one_hot_encode_example(self, example: dict) -> tuple:
+        """Takes an example having keys 'image' and 'label' and returns example
+        with keys 'image' and 'target'. 'target' is one hot encoded.
         Args:
-            image: tensor of the image
-            scale_size: Size of image to scale to
-            final_size: Size of final image
-
+            example: an example having keys 'image' and 'label'
         Returns:
-            Tensor of shape (final_size, final_size, 3)
+            example having keys 'image' and 'target'
```

Comment:
If `example` has a special class like `tf.train.Example()` then mention that otherwise `example` -> `Example`.